### PR TITLE
Darwin rusage

### PR DESCRIPTION
**Pull request by fengb**
Tests against `time -l` results:

```
$ /usr/bin/time -l ./rusage
rusage{ .utime = timeval{ .tv_sec = 0, .tv_usec = 398281 },.stime = timeval{ .tv_sec = 0, .tv_usec = 270910 },
.maxrss = 843776, .ixrss = 0, .idrss = 0, .isrss = 0,.minflt = 226, .majflt = 0, .nswap = 0, .inblock = 0,
.oublock = 0, .msgsnd = 0, .msgrcv = 0,.nsignals = 0, .nvcsw = 1, .nivcsw = 100161 }

        5.22 real         0.39 user         0.27 sys
    860160  maximum resident set size
         0  average shared memory size
         0  average unshared data size
         0  average unshared stack size
       230  page reclaims
         0  page faults
         0  swaps
         0  block input operations
         0  block output operations
         0  messages sent
         0  messages received
         0  signals received
         1  voluntary context switches
    100161  involuntary context switches
```

### CODE REVIEW

File: lib/std/os/bits/linux.zig [hunk 1]
```diff
@@ -1769,7 +1769,7 @@ pub const rusage = extern struct {
     utime: timeval,
     stime: timeval,
     maxrss: isize,
-    ix_rss: isize,
+    ixrss: isize,
```
Comment (fengb) [inline lib/std/os/bits/linux.zig:1772]: @data-man I'm not sure why this had an underscore. Do you mind if it's removed?

File: lib/std/os/bits/darwin.zig [hunk 2]
```diff
@@ -1474,3 +1474,25 @@ pub const CLOCK_UPTIME_RAW = 8;
...
 pub const CLOCK_PROCESS_CPUTIME_ID = 12;
 pub const CLOCK_THREAD_CPUTIME_ID = 16;
+
+pub const RUSAGE_SELF = 0;
+pub const RUSAGE_CHILDREN = -1;
+
+pub const rusage = extern struct {
+    utime: timeval,
+    stime: timeval,
+    maxrss: isize,
```
Comment (daurnimator) [inline lib/std/os/bits/darwin.zig:1484]: are these really signed? What would a negative maxrss mean?

Comment (fengb) [reply lib/std/os/bits/darwin.zig:1484]: They're defined as `long` in [man pages](https://opensource.apple.com/source/man/man-151/usr/man/man2/getrusage.2.auto.html) and [headers](https://github.com/apple/darwin-xnu/blob/master/bsd/sys/resource.h#L187-L201)

These match the existing Linux definitions, which are also longs.

File: lib/std/os/bits/linux.zig (hunk 1 above)
Comment (data-man) [reply lib/std/os/bits/linux.zig:1772]: @fengb 
Ok.

**Pull request merged**